### PR TITLE
Feat/auto accept trusted users

### PR DESCRIPTION
**Pull request by Ldoppea**
This PR:
- Implements a dialog to show users that are waiting for 2-steps confirmation on user's login
image
- Implements users auto-confirmation after creating a sharing if recipient is a trusted user
- Refactor sharing method into a new sharing service

____

TODO:
- [ ] bump cozy-ui version after merging https://github.com/cozy/cozy-ui/pull/1905
- [ ] bump cozy-sharing version after https://github.com/cozy/cozy-libs/pull/1353

### CODE REVIEW

File: src/cozy/services/sharing.service.ts [hunk 1]
```diff
@@ -143,6 +174,32 @@ export class SharingService {
         return organizationsWithoutCozy;
     }
 
+    protected async loadTrustedUsers() {
+        const queryDefinition = Q('com.bitwarden.contacts')
+            .select(['email', 'public_key', 'confirmed'])
+            .where({confirmed: true});
```
Comment (Crash--) [inline src/cozy/services/sharing.service.ts:182]: Do not hesitate to add an `indexFields` sorted by the most restrictive field first https://docs.cozy.io/en/tutorials/data/queries/#index-fields

File: src/cozy/wrappers/sharing/sharing.component.ts [hunk 2]
```diff
@@ -160,4 +170,19 @@ export class SharingComponent extends AngularWrapperComponent {
...
+
+    protected async onShare({
+        document,
+        recipients,
+        readOnlyRecipients,
+    }: OnShareEventArgs) {
+        const recipientsToConfirm = [...recipients, ...readOnlyRecipients];
+        const organizationId = document.id;
+
+        this.sharingService.autoConfirmTrustedUsers(organizationId, recipientsToConfirm);
```
Comment (Crash--) [inline src/cozy/wrappers/sharing/sharing.component.ts:189]: no need to await?

**Review by Crash-- — APPROVED** [review]
looks good

File: src/cozy/services/sharing.service.ts (hunk 1 above)
Comment (Ldoppea) [reply src/cozy/services/sharing.service.ts:182]: After reading the `indexFields` documentation:
- `confirmed` is not always present in `com.bitwarden.contacts`
  - My understanding is that I can create and index on `confirmed`, the requirement is that I don't use `$exists` on it. Is that correct?
- There is no need to add `indexField` for fields that are not present in the `where` condition. Do you confirm that?

Comment (Crash--) [reply src/cozy/services/sharing.service.ts:182]: If the field is not always present, then you can read the documentation about partial indexes https://docs.cozy.io/en/tutorials/data/advanced/#partial-indexes :D 

> There is no need to add indexField for fields that are not present in the where condition. Do you confirm that?

yep

Comment (Ldoppea) [reply src/cozy/services/sharing.service.ts]: `.indexFields(["confirmed"])` has been added in last forced push

File: src/cozy/wrappers/sharing/sharing.component.ts (hunk 2 above)
Comment (Ldoppea) [reply src/cozy/wrappers/sharing/sharing.component.ts]: Not necessary as it is just a callback but you're right. I added the `await` and I changed cozy-sharing to await the callback to finish (PR not created yet)

**Pull request merged**